pkg/config/msp_test.go

```diff
@@ -947,24 +947,57 @@ func TestCreateApplicationMSPCRL(t *testing.T) {
 		ChannelGroup: channelGroup,
 	}
 
-	c := New(config)
+	originalConfigTx := New(config)
 
-	org1MSP, err := c.ApplicationMSP("Org1")
+	// update org2MSP to include an intemediate cert that is different
+	// from the root cert
+	org2MSP, err := originalConfigTx.ApplicationMSP("Org2")
 	gt.Expect(err).NotTo(HaveOccurred())
-
-	cert, privKey, _ := certPrivKeyCRL(org1MSP)
-	certToRevoke, _ := generateCertAndPrivateKeyFromCACert(t, "org1.example.com", cert, privKey)
-	signingIdentity := &SigningIdentity{
-		Certificate: cert,
-		PrivateKey:  privKey,
-		MSPID:       "MSPID",
-	}
-	newCRL, err := c.CreateApplicationMSPCRL("Org1", signingIdentity, certToRevoke)
+	org2Cert, org2PrivKey, _ := certPrivKeyCRL(org2MSP)
+	org2IntermediateCert, org2IntermediatePrivKey := generateIntermediateCACertAndPrivateKey(t, "org2.example.com", org2Cert, org2PrivKey)
+	org2MSP.IntermediateCerts = append(org2MSP.IntermediateCerts, org2IntermediateCert)
+	err = originalConfigTx.UpdateApplicationMSP(org2MSP, "Org2")
 	gt.Expect(err).NotTo(HaveOccurred())
-	gt.Expect(newCRL.TBSCertList.RevokedCertificates).To(HaveLen(1))
-	gt.Expect(newCRL.TBSCertList.RevokedCertificates[0].SerialNumber).To(Equal(certToRevoke.SerialNumber))
 
-	// TODO create a CRL using an intermediate cert
+	t.Run("create CRL using a root cert", func(t *testing.T) {
```

Comment:
nit: can we table test these? seems like ultimately they're both testing using a different certToRevoke which can be set up beforehand pretty easily. Alternatively you could just combine both of them into one test that uses the same setup and just does one check after the other on the same config.